metadata/metadata.c

```diff
@@ -415,17 +433,17 @@ void _add_virtual_disk_range_to_virtual_disk(VirtualDisk *virtual_disk, VirtualD
     virtual_disk->n_ranges = last + 1;
 }
 
-VirtualDisk *create_virtual_disk(Metadata *metadata, char *name, uint64_t size, Allocator *allocator)
+VirtualDisk *create_virtual_disk(Metadata *metadata, char *name, uint64_t size)
```

Comment:
The goal of hiding the allocator from the user would be already achieved by removing the `allocator` parameter from the `create_virtual_disk` function only. I would still pass it as a parameter to all the underscore functions since it might come in handy once we do want to support different allocators for different virtual disks.